qa/util/src/main/java/io/camunda/qa/util/multidb/CamundaMultiDBExtension.java

```diff
@@ -301,34 +310,121 @@ public void beforeAll(final ExtensionContext context) {
+  private KeycloakContainer setupKeycloak() {
+    keycloakContainer = DefaultTestContainers.createDefaultKeycloak();
+    closeables.add(keycloakContainer);
+    keycloakContainer.start();
+
+    final var realm = new RealmRepresentation();
+    realm.setRealm(KEYCLOAK_REALM);
+    realm.setEnabled(true);
+    try (final var keycloak = keycloakContainer.getKeycloakAdminClient()) {
+      keycloak.realms().create(realm);
+    }
+    setupUserInKeycloak(
+        TestStandaloneBroker.DEFAULT_MAPPING_ID, TestStandaloneBroker.DEFAULT_MAPPING_CLAIM_VALUE);
+
+    return keycloakContainer;
+  }
+
+  private void createEntities(final Class<?> testClass, final Boolean shouldSetupKeycloak) {
     final var users = findUsers(testClass, null, ModifierSupport::isStatic);
+    final var mappings = findMappings(testClass, null, ModifierSupport::isStatic);
     final var groups = findGroups(testClass, null, ModifierSupport::isStatic);
     final var roles = findRoles(testClass, null, ModifierSupport::isStatic);
-    entityManager.withUser(users).withGroups(groups).withRoles(roles).await();
+    entityManager
+        .withUser(users)
+        .withMappings(mappings)
+        .withGroups(groups)
+        .withRoles(roles)
+        .await();
     users.forEach(
         user ->
             authenticatedClientFactory.createClientForUser(applicationUnderTest.application, user));
+
+    mappings.forEach(
+        mapping -> {
+          authenticatedClientFactory.createClientForMapping(
+              applicationUnderTest.application, mapping);
+
+          if (shouldSetupKeycloak) {
+            setupUserInKeycloak(mapping.id(), mapping.claimValue());
+          }
+        });
+  }
+
+  private void setupUserInKeycloak(final String mappingId, final String claimValue) {
+    final var clientRepresentation = new ClientRepresentation();
+    clientRepresentation.setClientId(claimValue);
+    clientRepresentation.setEnabled(true);
+    clientRepresentation.setClientAuthenticatorType("client-secret");
+    clientRepresentation.setSecret(claimValue);
+    clientRepresentation.setServiceAccountsEnabled(true);
+
+    final var userRepresentation = new UserRepresentation();
+    userRepresentation.setId(claimValue);
+    userRepresentation.setUsername(claimValue);
+    userRepresentation.setServiceAccountClientId(mappingId);
+    userRepresentation.setEnabled(true);
+
+    try (final var keycloak = keycloakContainer.getKeycloakAdminClient()) {
+      final var realm = keycloak.realm(CamundaMultiDBExtension.KEYCLOAK_REALM);
+      realm.clients().create(clientRepresentation).close();
+      realm.users().create(userRepresentation).close();
+    }
   }
 
   private void manageApplicationUnderTest() {
     final var application = applicationUnderTest.application;
     closeables.add(application);
     application.start();
-    application.awaitCompleteTopology(application.brokerConfig());
+    application.awaitCompleteTopology(
```

Comment:
I didn't read the logs as I'm not at my laptop but this is likely why other tests are failing now. Im not sure if there's always an admin client available.